scripts/prepare_heat_demand.py

```diff
@@ -0,0 +1,305 @@
+    if consumer + "_" + type in demands.columns:
+        demands[consumer + "_" + type] = np.sum(demands[consumer + "_" + type], [value])
+    else:
+        demands[consumer + "_" + type] = [value]
+    return demands
+
+
+def get_heat_demand(path, region, scenario):
+    """
+    This function returns ghd and hh demands together with their unit of a given region and a
+    given scenario
+
+    Parameters
+    ----------
+    path : str
+        input path
+    region : str
+        region e.g. BB as Brandenburg
+    scenario : str
+        scenario e.g. "base"
+
+    Returns
+    -------
+    demand_ghd : float
+        total demand ghd
+    demand_hh : float
+        total demand hh
+    demand_ghd_unit : str
+        unit of total demands
+
+    """
+    # Read state heat demands of ghd and hh sectors
+    sc = dp.load_b3_scalars(path)  # TODO: Add ', sep=";"' if from schema
+    consumers = ["ghd", "efh", "mfh"]
+    types = ["central", "decentral"]
+    demands = pd.DataFrame()
+    units = []
+
+    for row in sc.iterrows():
+        # Get heat demands in region
+        if (
+            "heat" in row[1]["carrier"]
+            and "demand" in row[1]["tech"]
+            and row[1]["scenario"] == scenario
+            and row[1]["region"] == region
+        ):
+            for consumer in consumers:
+                for type in types:
+                    if (
+                        consumer in row[1]["name"]
+                        and type in row[1]["name"]
+                        and "de" + type not in row[1]["name"]
+                    ):
+                        check_central_decentral(
+                            demands, row[1]["var_value"], consumer, type
+                        )
+                        units.append(row[1]["var_unit"])
+
+    if len(list(set(units))) != 1:
+        raise ValueError(
+            f"Unit mismatch in scalar data of heat demands. "
+            f"Please make sure units match in {path}."
+        )
+
+    return demands, list(set(units))
+
+
+def calculate_heat_load(region, scenario):
+    """
+    This function calculates a heat load profile of Industry, trade,
+    service (ghd: Gewerbe, Handel, Dienstleistung) and Household (hh: Haushalt)
+    sectors
+
+    Parameters
+    ----------
+    region : str
+        region e.g. BB as Brandenburg
+    scenario : str
+        scenario e.g. "base"
+
+    Returns
+    -------
+    heat_load_total : pd.DataFrame
+         DataFrame that contains the calculated total heat load
+
+    """
+    # Set examined years
+    years = np.arange(2010, 2020)
+
+    # Read time series template
+    ts_template = dp.load_b3_timeseries(in_path5, sep=";")
+
+    # Add empty data frame for results / output
+    heat_load = pd.DataFrame(columns=ts_template.columns)
+
+    for index, year in enumerate(years):
+        # Get holidays
+        holidays = get_holidays(in_path3, year, region)
+
+        # Read weather data
+        path_weather_data = os.path.join(
```

Comment:
I advise to list the path to the directory containing weather data among the inputs in the snakemake rule. It is already there, but only as `raw`.

Reply:
Done with commits https://github.com/rl-institut/oemof-B3/pull/76/commits/cf677c1b9cb7e06da99baebc17180e89fe89465f and https://github.com/rl-institut/oemof-B3/pull/76/commits/f2b37f8082de1b48b5c269947c5a8171703cf94b.